packages/orama/src/components/sorter.ts

```diff
@@ -103,56 +101,99 @@ async function insert(
   if (!sorter.enabled) {
     return
   }
+
+  sorter.language = language
+  sorter.isSorted = false
+
   const s = sorter.sorts[prop]
 
-  let predicate: (value: [string, SortValue]) => boolean
-  switch (schemaType) {
+  s.docs.set(id, s.orderedDocs.length);
+  s.orderedDocs.push([id, value]);
+}
+
+function ensureIsSorted(sorter: Sorter): void {
+  if (sorter.isSorted) {
+    return
+  }
+
+  if (!sorter.enabled) {
+    return
+  }
+
+  for (const prop of Object.keys(sorter.sorts)) {
+    ensurePropertyIsSorted(sorter, prop);
+  }
+
+  sorter.isSorted = true;
+}
+
+function stringSort(language: string | undefined, value: [string, SortValue], d: [string, SortValue]): number {
+  return (value[1] as string).localeCompare(d[1] as string, language)
+}
+
+function numerSort(value: [string, SortValue], d: [string, SortValue]): number {
+  return (value[1] as number) - (d[1] as number)
+}
+
+function booleanSort(value: [string, SortValue], d: [string, SortValue]): number {
+  return d[1] as boolean ? -1 : 1
+}
+
+function ensurePropertyIsSorted(sorter: Sorter, prop: string): void {
+  const s = sorter.sorts[prop];
+
+  let predicate: (value: [string, SortValue], d: [string, SortValue]) => number
+  switch (s.type) {
     case 'string':
-      predicate = stringSort.bind(null, value, language)
+      predicate = stringSort.bind(null, sorter.language)
       break
     case 'number':
-      predicate = numerSort.bind(null, value)
+      predicate = numerSort.bind(null)
       break
     case 'boolean':
-      predicate = booleanSort.bind(null, value)
+      predicate = booleanSort.bind(null)
       break
   }
 
-  // Find the right position to insert the element
-  let index = s.orderedDocs.findIndex(predicate)
-  if (index === -1) {
-    index = s.orderedDocs.length
-    s.orderedDocs.push([id, value])
-  } else {
-    s.orderedDocs.splice(index, 0, [id, value])
-  }
-  s.docs[id] = index
+  s.orderedDocs.sort(predicate);
 
   // Increment position for the greather documents
-  const orderedDocsLength = s.orderedDocs.length
-  for (let i = index + 1; i < orderedDocsLength; i++) {
+  const orderedDocsLength = s.orderedDocs.length;
+  for (let i = 0; i < orderedDocsLength; i++) {
     const docId = s.orderedDocs[i][0]
-    s.docs[docId]++
+    s.docs.set(docId, i);
+  }
+}
+
+function ensureOrderedDocsAreDeleted(sorter: Sorter): void {
+  for (const prop of Object.keys(sorter.sorts)) {
```

Comment:
Move `Object.keys` outside the loop condition to avoid accessing the object keys on every iteration